jupyterlab_caip_optimizer/src/components/sidebar_widget.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import { ReactWidget } from '@jupyterlab/apputils';
+import { Provider, connect } from 'react-redux';
+import {
+  Box,
+  Button,
+  Table,
+  TableHead,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableRow,
+  Typography,
+  CircularProgress,
+} from '@material-ui/core';
+import { RootState, AppDispatch } from '../store/store';
+import { Study } from '../types';
+import { setView } from '../store/view';
+import { Store } from 'redux';
+import { prettifyStudyName } from '../service/optimizer';
+import { style } from 'typestyle';
+
+const sidebarStyle = style({
+  height: '100%',
+  overflow: 'scroll',
+});
+
+const rowStyle = style({
+  $nest: {
+    '&:hover': {
+      cursor: 'pointer',
+    },
+  },
+});
+
+const mapStateToProps = (state: RootState) => ({
+  loading: state.studies.loading,
+  error: state.studies.error,
+  studies: state.studies.data,
+});
+
+const mapDispatchToProps = (dispatch: AppDispatch) => ({
+  openDashboard: () => dispatch(setView({ view: 'dashboard' })),
+  openStudy: (studyName: string) =>
+    dispatch(setView({ view: 'studyDetails', studyId: studyName })),
+});
+
+interface SidebarProps {
+  error?: string;
+  loading: boolean;
+  studies?: Study[];
+  openDashboard: () => void;
+  openStudy: (studyName: string) => void;
+}
+
+export const Sidebar = ({
+  studies,
+  error,
+  loading,
+  openDashboard,
+  openStudy,
+}: SidebarProps) => {
+  return (
+    <Box
+      height={1}
+      width={1}
+      bgcolor="white"
+      borderRadius={0}
+      display="flex"
+      flexDirection="column"
+      className={sidebarStyle}
+    >
+      <Box display="flex" pt={2}>
+        <Box mx="auto">
+          <Typography variant="h4" gutterBottom>
+            Optimizer
+          </Typography>
+
+          <Button color="primary" onClick={() => openDashboard()}>
+            Main Dashboard
+          </Button>
+        </Box>
+      </Box>
+
+      {!!error && <Typography color="error">{error}</Typography>}
+      {loading && <CircularProgress />}
+      {!!studies && studies.length > 0 && (
```

Comment:
Maybe we should display text (e.g. "No studies found") if it's an empty list?

Reply:
Here's what it looks like now: ![image](https://user-images.githubusercontent.com/12939414/86960556-ef85f480-c11c-11ea-9a1c-01af91ca3e75.png)

Reply:
Not sure if we'd want to keep the error message in if there are correctly no studies loaded (feel free to address this in a separate PR)